src/main/java/io/cdap/plugin/servicenow/source/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -48,7 +49,11 @@
   private static final String FIELD_CREATED_ON = "sys_created_on";
   private static final String FIELD_UPDATED_ON = "sys_updated_on";
   private static final String OAUTH_URL_TEMPLATE = "%s/oauth_token.do";
-
+  private static final ImmutableMap<Integer, Integer> retryDelay = ImmutableMap.of(
```

Comment:
Use a library instead of implementing retry/backoff logic yourself.
Example - https://github.com/data-integrations/google-cloud/pull/918/files#diff-6a7f8b65e9e06b9f3d93abd2603ec319c4e9c5b6083fcadac773b32f55e1da8cR318

Reply:
Couldn't find the reference to the library related to retry on the link shared by you.

Can we have a short call for this?

Reply:
Try using https://github.com/rholder/guava-retrying.

Reply:
Thank you for sharing the library. The dev team is done with the implementation using the guava-retry library and is currently unit testing the code. This will be followed by functional testing by the QA team which will likely take a day or two more to complete. 

As this PR, containing a fix for the data loss issue happens to be a time-sensitive deliverable, part of the ServiceNow Release, can we raise a separate PR for the guava-retry implementation and let this one get through?

Reply:
Retry implemented using Guava-retrying library [150737e](https://github.com/data-integrations/servicenow-plugins/pull/16/commits/150737e8afe7681df7a7356fea149043d9c98075)